### PR TITLE
fix: LabelEncoder params consistent with Sklearn

### PR DESCRIPTION
**Pull request by ashleyxuu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/ml/base.py [hunk 1]
```diff
@@ -195,3 +195,23 @@ def fit_transform(
...
+    def fit(self, y):
+        pass
+
+    @abc.abstractmethod
+    def transform(self, y):
+        pass
+
+    def fit_transform(
+        self,
+        y: Optional[Union[bpd.DataFrame, bpd.Series]] = None,
```
Comment (GarrettWu) [inline bigframes/ml/base.py:215]: It can't be None.

File: bigframes/ml/base.py [hunk 2]
```diff
@@ -195,3 +195,23 @@ def fit_transform(
         y: Optional[Union[bpd.DataFrame, bpd.Series]] = None,
     ) -> bpd.DataFrame:
         return self.fit(X, y).transform(X)
+
+
+class LabelTransformer(BaseEstimator):
+    """A BigQuery DataFrames Transformer base class that transforms data.
```
Comment (GarrettWu) [inline bigframes/ml/base.py:201]: More specific only to labels.

File: bigframes/ml/base.py [hunk 3]
```diff
@@ -212,6 +212,6 @@ def transform(self, y):
 
     def fit_transform(
         self,
-        y: Optional[Union[bpd.DataFrame, bpd.Series]] = None,
+        y: Optional[Union[bpd.DataFrame, bpd.Series]],
```
Comment (GarrettWu) [inline bigframes/ml/base.py:215]: remove Optional

**Review by GarrettWu — CHANGES_REQUESTED** [review]

File: bigframes/ml/base.py (hunk 3 above)
Comment (ashleyxuu) [reply bigframes/ml/base.py]: My bad. Fixed now.

**Review by GarrettWu — APPROVED** [review]
